DoSomethingArchive/legacy-website
# Restructuring preprocess.inc code.

#### What's this PR do?

This is an attempt to restructure the unwieldy **preprocess.inc** file into something more usable by breaking it down into smaller preprocess files that deal with specific entities (node, page, taxonomy, user, etc). Moving forward new preprocess function can be added to the respective file, or if a new entity, then just add an include file and celebrate some sense of order in this chaotic world we live in.
#### Where should the reviewer start?

Probably at **preprocess.inc** and head to the broken out files from there. Entirely a cut and paste job, so not much code change; just double check the `require_once()`'s look correct.
#### How should this be manually tested?

Pull down the branch, clear cache and load up the site. Make sure nothing is broken. I've tested this on campaign pages and no errors/notices show up.
#### Any background context you want to provide?

I'm working on upgrading the reportback gallery on campaign pages to use our updated API endpoints, and working with the preprocess file has always been a bit nutty with the crazy number of functions there.

---

@DFurnes @sbsmith86 @angaither
